src/lib/storage/buffer/persistence_manager.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <fcntl.h>
+#include <filesystem>
+#include <mutex>
+#include <unordered_map>
+#include "page_id.hpp"
+#include "types.hpp"
+
+namespace hyrise {
+
+/**
+ * PersistenceManager manages read and write access of pages to an SSD. PersistenceManager can either use a block device directly by assigning equal-sized regions
+ * to each page size type or use a directory with one file for each page size type. Files are opened using direct IO (O_DIRECT) to skip the OS page cache. 
+ * This requires read and write buffers to be aligned to the page boundary (e.g. 4KB). Destructing the PersistenceManager closes all open file descriptors and 
+ * deletes all files in the directory. Block devices are not cleared.
+ * 
+ * Writing and reading concurrently to different pages is thread-safe. Accessing the same page should be guarded with concurrency primitives accordingly. 
+ * We usually use the exclusive latching defined by the Frame class. Currently, we only support synchronous IO, which means that the calling thread is 
+ * blocked until the IO operation has finished. Employing aio, libaio or io_uring is considered future work, but may lead to significant performance improvements.
+ */
+class PersistenceManager final : public Noncopyable {
+ public:
+  // The mode can be either a block device or a directory with one file for each page size type.
+  enum class Mode { BLOCK, FILE_PER_SIZE_TYPE };
+
+  // Creates a new SSDRegion. The path can either be a directory or a block device. If the path is a directory, a file is created for each page size type.
+  PersistenceManager(const std::filesystem::path& path);
+
+  // Destroy the SSDRegion and close all open file descriptors. This also deletes all files in the directory. Block devices are not cleared.
+  ~PersistenceManager();
+
+  // Writes the given data to the page with the given page id. The buffer address must be aligned to the page size boundary.
+  void write_page(const PageID page_id, std::byte* data);
```

Comment:
I think this should be the only case here for us to use raw pointers since we interact with raw memory – same with allocators. Shared Pointers do not make sense since they include ownership.

Reply:
from the link above:

> This sounds like a case where you'd want to pass a raw pointer instead of a shared_ptr. –